java/src/main/java/ai/onnxruntime/OnnxRuntime.java

```diff
@@ -78,6 +78,9 @@ final class OnnxRuntime {
   /** Tracks if the shared providers have been extracted */
   private static final Set<String> extractedSharedProviders = new HashSet<>();
 
+  /** The OrtBaseApi handle. Only needed for direct registration of custom ops. */
+  static long ortBaseHandle;
```

Comment:
@Craigacp what's the best way to make this publicly accessible? OnnxRuntime is not a public class and I don't want to make it public just for this. I could maybe add a getter to Utils.java, but that doesn't have a static init blck to call `OnnxRuntime.init()` and I'm not sure if it's fine to add that block there.

Reply:
I've tried not to expose any `long`s which store pointers to users, and so I'd really strongly prefer to keep this package private. Could we expose a method on `OrtSession.SessionOptions` which accepts a function pointer to the `OrtStatus* RegisterCustomOps(OrtSessionOptions * options, const OrtApiBase* api)` call for that library, then supplies the appropriate arguments to it?

I'm also not sure how the lifetime of the loaded custom ops dll is supposed to be handled, on the Java side we keep track of all those libraries in the `SessionOptions` and free them when the options is closed. What's the appropriate way to do that for this use case?

Reply:
This is somewhat of a special-case usage where you're linking against onnxruntime and a library that provides custom ops (onnxruntime-extensions in my case, but theoretically could be a different one), so there's no dynamic loading of the custom ops. The user still needs to connect the two up at runtime, but ORT doesn't need to load or care about the lifetime of the custom ops library.

Providing a function pointer to a SessionOptions instance sounds like a good approach. @yuslepukhin could the V2 for custom ops registration in the ORT API support this (additional variant taking function pointer instead of path)? That would make things more consistent.

Reply:
Ah ok. If we're willing to accept a little more work on the part of the developer who wants to link these things up, then we could make a base class in ORT Java which is subclassed by developers who want to add in a custom op library, then they init their subclass, it bounces through to our base class which static inits ORT and then calls back to their code to get the pointer which we pass into the JNI. We could then ensure that the pointer is never visible in user code, which is where I'd like to be.

Reply:
> This is somewhat of a special-case usage where you're linking against onnxruntime and a library that provides custom ops (onnxruntime-extensions in my case, but theoretically could be a different one), so there's no dynamic loading of the custom ops. The user still needs to connect the two up at runtime, but ORT doesn't need to load or care about the lifetime of the custom ops library.
> 
> Providing a function pointer to a SessionOptions instance sounds like a good approach. @yuslepukhin could the V2 for custom ops registration in the ORT API support this (additional variant taking function pointer instead of path)? That would make things more consistent.

We will see what we can do.

Reply:
I'll play around with that approach and see if I hit any issues.